crates/runtime/README.md

```diff
@@ -0,0 +1,662 @@
+allowing it to maintain a consistent view of the execution environment
+throughout the lifetime of a function call.
+
+One of the fundamental responsibilities of host functions is memory management.
+They provide methods for reading from and writing to the guest's memory space,
+ensuring that all memory accesses are bounds-checked and secure. This includes
+functions for reading raw bytes from memory as well as higher-level operations
+like retrieving UTF-8 strings. By mediating all memory interactions, the host
+functions prevent potential security vulnerabilities that could arise from
+unchecked memory access.
+
+Register operations form another critical aspect of the host functions. They
+allow the guest code to store and retrieve data in a temporary storage system,
+which is particularly useful for passing information between different parts of
+the Wasm module or for storing intermediate results. The host functions ensure
+that all register operations adhere to the size and count limits specified in
+`VMLimits`, preventing resource abuse and maintaining the integrity of the
+execution environment.
+
+Storage interactions are also facilitated through host functions. These
+functions provide a key-value storage interface, allowing the guest code to read
+from and write to a persistent storage system. By channeling all storage
+operations through these functions, the runtime can enforce access controls,
+manage resource usage, and potentially implement advanced features like rollback
+or versioning.
+
+Logging and event emission are crucial for debugging and monitoring Wasm
+execution. The host functions provide methods for the guest code to log messages
+and emit events, which are collected by the runtime for later analysis. These
+functions ensure that logging and event emission adhere to the specified limits,
+preventing potential denial-of-service attacks through excessive logging.
+
+One of the more powerful capabilities provided by the host functions is the
+ability to make external HTTP requests. The `fetch()` function allows the guest
+code to interact with external services, opening up possibilities for complex,
+interconnected applications. However, this function is implemented with careful
+consideration for security, potentially including restrictions on allowed URLs
+or rate limiting to prevent abuse.
+
+Error handling and panic management are also key responsibilities of the host
+functions. They provide mechanisms for the guest code to signal errors or panic
+conditions, which are then safely propagated to the runtime environment. This
+includes functions for reporting panics with detailed location information,
+allowing for better debugging and error reporting.
+
+Cryptographic operations, such as keypair creation, are also made available
+through host functions. These provide secure, audited implementations of
+cryptographic primitives to the guest code, ensuring that sensitive operations
+are performed correctly and securely.
```

Comment:
We don't have this

Reply:
I must say I am surprised - it was touched on in another thread, but is it a design choice that we will _not_ provide this? I expected us to be able to. Will we always require a user to supply a keypair? Surely it would ease the process if we allow the option to create one? I'm just keen to understand more about this 🙂

Reply:
the executor's identity is already defined once they join the context, and that's the only identity that matters wrt the runtime

any other keypairs you want to create can be done within the guest environment like we currently do with `rock-paper-scissors`

https://github.com/calimero-network/rock-paper-scissors/blob/de166ecf11d41e963df5facb8a2457edc1bf6320/logic/rock-paper-scissors/src/lib.rs#L63

Although it's use in RPS was an accommodation for lacking runtime identities, which is changing